### ISSUE TITLE
upbound-agent CrashLoopBackOff after token removal

### ISSUE THREAD

**Issue by ytsarev** [upbound/universal-crossplane#236]
### What happened?

After removal of API token in UpboundCloud profile, the upbound-agent pod continuously crashes with

```sh
upbound-agent-d98d5f59d-pphhs                                 0/1     CrashLoopBackOff   1 (5s ago)    7s
 k -n upbound-system logs -f upbound-agent-d98d5f59d-pphhs
upbound-agent: error: failed to fetch public certs: agent certs request failed with 401 Unauthorized - {"message":"failed to verify token: sql: no rows in result set"}
```

### How can we reproduce it?
* Create API token in UbC
* `up login` using this token
* Attach UXP instance with `up ctp attach $ctp-name | up uxp connect -`
* Observe that `upbound-agent` pod is getting created and is in Running state
* Delete API Token in UbC
* Force recreation of upbound-agent pod by deleting it
* Observe CrashLoopBackoff

The situation has a worse impact because even after creating a new token and the relogin it will block the attachment of new UXP instances until the CrashLoopBackof is forcefully deleted again. Only after that, it will pick up a new valid auth token.

### What environment did it happen in?
```
universal-crossplane-1.6.3-up.1	1.6.3-up.1
```

**Comment by jeanduplessis** [comment]
Closing as no longer relevant
